### ISSUE TITLE
Add "Leave Thread" Option

### ISSUE THREAD

**Issue by HenrickTheBull** [bluesky-social/social-app#3445]
**Is your feature request related to a problem? Please describe.**

The feature request is related to the problem of being part of a thread and no longer wishing to partake in the thread.

**Describe the solution you'd like**

An option much like Twitter where you can remove yourself from the thread which would be the hardest mute possible for a thread and it would disallow people from tagging you in the thread again.

**Describe alternatives you've considered**

There isn't really any alternatives to this. Twitter had something here I think Bluesky could replicate.

**Additional context**

![image](https://github.com/bluesky-social/social-app/assets/3945192/88c4957b-8a11-457f-951c-7d46f6b4c9de)
https://help.twitter.com/en/using-x/mentions-and-replies#mention-control

**Comment by cooljeanius** [comment]
Yeah, this feature was pretty useful on Twitter; "Mute Thread" is a helpful half-step, but on Twitter "Mute Thread" and "Leave Thread" were kept as separate options, so having them both separately on BlueSky would be useful, too.
